pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go

```diff
@@ -155,6 +145,81 @@ func (cs *cpuServer) ListAndWatch(_ *advisorsvc.Empty, server cpuadvisor.CPUAdvi
 	}
 }
 
+func (cs *cpuServer) assembleResponse(result *types.InternalCPUCalculationResult) (*cpuadvisor.ListAndWatchResponse, error) {
+	calculationEntriesMap := make(map[string]*cpuadvisor.CalculationEntries)
+	blockID2Blocks := NewBlockSet()
+	cs.assemblePoolEntries(result, calculationEntriesMap, blockID2Blocks)
+
+	// assmble per-numa headroom
```

Comment:
headroom is not equal to cpu provison(reclaimed_cores pool size), i think headroom should not be considerd here.